ts/utils/idps.ts

```diff
@@ -1,76 +1,130 @@
-import { SpidIdp } from "../../definitions/content/SpidIdp";
+import { ImageSourcePropType } from "react-native";
+import { SpidIdp as GeneratedSpidIdpType } from "../../definitions/content/SpidIdp";
+
+export const fromGeneratedToLocalSpidIdp = (
+  idps: ReadonlyArray<GeneratedSpidIdpType>
+): ReadonlyArray<SpidIdp> =>
+  idps.map(idp => ({
+    id: idp.id,
+    name: idp.name,
+    logo: {
+      light: { uri: idp.logo },
+      dark: { uri: idp.logoDark }
```

Comment:
Here we're assigning the `dark` property even when no value is provided.
This results in an unintended behavior: the logic doesn't fall back to the light version of the logo as expected.

Is that correct?

<img src="https://github.com/user-attachments/assets/c8072208-c112-4640-8895-2f1141101b5d" width="200" />

Reply:
<img src="https://github.com/user-attachments/assets/478c3928-e5f7-4153-90d5-649a09552df0" >

Reply:
@shadowsheep1 You're right. Just replaced with the following condition, also updated the relevant tests:
```jsx
...(idp.logoDark ? { dark: { uri: idp.logoDark } } : {})
```